examples/simple-server.py

```diff
@@ -5,6 +5,24 @@
 from callosum.lower.zeromq import ZeroMQAddress, ZeroMQTransport
 
 
+# add the following handler when you want to test how
+# cancellation on invocation timeout works
+'''
+async def handle_echo(request):
+    try:
+        print("Before async sleep")
+        await asyncio.sleep(1000)
+        print("After async sleep")
+        return {
+            'received': request.body['sent'],
+        }
+    except asyncio.CancelledError:
+        print("Task was cancelled successfully")
+        # NOTE: due to strange behaviour of asyncio, I have to reraise
+        # otherwise, the task.cancelled() returns False
```

Comment:
Good catch. Unless your are at the top-level coroutine of a task that explicitly handles resource clean up upon cancellation, you must re-raise any `asyncio.CancelledError`.
In Python 3.8, `asyncio.CancelledError` will become a `BaseException`, not `Exception` to encourage this practice.